src/charts/common/helpers.js

```diff
@@ -0,0 +1,76 @@
+import {select} from 'd3-selection';
+
+const validateConfiguration = (chart, configurationProperties) => {
+    let supportedMethods = Object.keys(chart.prototype.constructor);
+
+    let notSupportedMethods = configurationProperties
+        .filter((methodName) => !supportedMethods.includes(methodName));
+
+    if (notSupportedMethods.length) {
+        throw new Error(`Method not supported by Britechart: ${notSupportedMethods.join(' ')}`);
+    }
+};
+
+const validateContainer = (container) => {
+    if (container.empty()) {
+        throw Error('A root container is required');
+    }
+};
+
+const validateData = (data) => {
+    if (!data || !data.length) {
+        throw Error('Data must be defined');
+    }
+};
+
+export default function wrapChart(britechart) {
```

Comment:
This  looks great, but I think we are optimizing prematurely, as we still haven't figure out all the event related logic.
I would unroll this abstraction for now.

Reply:
Sure, I can unroll this for now until after we have a better idea of how to design it. My thinking was that right now we know the legend and stacked area both can use the same code for the most part. I just worry that after adding a few more vis types, we'll forget which ones use the same code and which ones are different, and it will be a bit more work to figure out when it is time to extract the shared code. Not the end of the world, though.

Reply:
Yeah, that could happen, but we will probably try to extract code in a more granular basis. Either way, we'll see soon!